src/main/java/org/traccar/database/OpenIdProvider.java

```diff
@@ -69,28 +77,51 @@ public class OpenIdProvider {
     private URI userInfoUrl;
     private URI baseUrl;
     private final String adminGroup;
+    private final String allowGroup;
 
     private LoginService loginService;
 
     @Inject
-    public OpenIdProvider(Config config, LoginService loginService) {
-        this.loginService = loginService;
-
-        force = config.getBoolean(Keys.OPENID_FORCE);
-        clientId = new ClientID(config.getString(Keys.OPENID_CLIENTID));
-        clientAuth = new ClientSecretBasic(clientId, new Secret(config.getString(Keys.OPENID_CLIENTSECRET)));
-
-        try {
-            callbackUrl = new URI(config.getString(Keys.WEB_URL, "") + "/api/session/openid/callback");
-            authUrl = new URI(config.getString(Keys.OPENID_AUTHURL, ""));
-            tokenUrl = new URI(config.getString(Keys.OPENID_TOKENURL, ""));
-            userInfoUrl = new URI(config.getString(Keys.OPENID_USERINFOURL, ""));
-            baseUrl = new URI(config.getString(Keys.WEB_URL, ""));
+    public OpenIdProvider(
+        Config config, LoginService loginService, HttpClient httpClient, ObjectMapper objectMapper
+        ) throws InterruptedException, IOException {
+            this.loginService = loginService;
+
+            force = config.getBoolean(Keys.OPENID_FORCE);
+            clientId = new ClientID(config.getString(Keys.OPENID_CLIENT_ID));
+            clientAuth = new ClientSecretBasic(clientId, new Secret(config.getString(Keys.OPENID_CLIENT_SECRET)));
+
+            try {
+                callbackUrl = new URI(config.getString(Keys.WEB_URL, "") + "/api/session/openid/callback");
+                baseUrl = new URI(config.getString(Keys.WEB_URL, ""));
```

Comment:
We have this logic here to constructing default URL:

https://github.com/traccar/traccar/blob/master/src/main/java/org/traccar/MainModule.java#L390-L401

Maybe we can extract it into some common place and reuse here as well?

Reply:
Did you have anywhere specific in mind? None of the helper files seem to be an obvious choice to me

Reply:
Maybe add it to the `ServletHelper`? We can also rename it to something more generic, like `WebHelper`.

Reply:
Something like that?